### DISCUSSION TITLE
Collision avoidance & stairs climbing functionalities

### DISCUSSION THREAD

**Discussion by RiccardoZuppetti** [abizovnuralem/go2_ros2_sdk#97] [General]
Hi, regarding the work done I was wondering if the project:

- already implements within it an obstacle avoidance module for navigation

- allows the to quadruped to climb stairs

As far as climbing some steps is concerned, I've already tried it but it doesn't seem to be able to do it by default, is it necessary to activate some mode for the robot before actually making it move in its surroundings?

**Comment by abizovnuralem** [comment]
Hello @RiccardoZuppetti , thanks for your questions:
1. If you control the dog using this SDK the obstacle avoidance module will not work
2. You can change the mode using webrtc to "climb stairs" and try, to find this functionality you can search in official unitree docs

Good luck!

**Reply by RiccardoZuppetti** [reply]
> 2. You can change the mode using webrtc to "climb stairs" and try, to find this functionality you can search in official unitree docs

@abizovnuralem thanks, I'll definitely try in the next coming days!

**Comment by RiccardoZuppetti** [comment]
Hi @abizovnuralem, in order to run everything is it required to have a machine with an Nvidia Gpu?
